subsys/bluetooth/Kconfig.logging

```diff
@@ -0,0 +1,1083 @@
+	select DEPRECATED
+	help
+	  Use this option to enable advertising debug logs for
+	  the Bluetooth mesh functionality.
+
+config BT_MESH_DEBUG_LOW_POWER
+	bool "[DEPRECATED] Low Power debug"
+	select DEPRECATED
+	help
+	  Use this option to enable Low Power debug logs for the
+	  Bluetooth mesh functionality.
+
+config BT_MESH_DEBUG_FRIEND
+	bool "[DEPRECATED] Friend debug"
+	select DEPRECATED
+	help
+	  Use this option to enable Friend debug logs for the
+	  Bluetooth mesh functionality.
+
+config BT_MESH_DEBUG_PROXY
+	bool "[DEPRECATED] Proxy debug"
+	select DEPRECATED
+	depends on BT_MESH_GATT
+	help
+	  Use this option to enable Proxy protocol debug logs.
+
+config BT_MESH_DEBUG_SETTINGS
+	bool "[DEPRECATED] Persistent settings debug"
+	select DEPRECATED
+	depends on BT_SETTINGS
+	help
+	  Use this option to enable persistent settings debug logs.
+
+config BT_MESH_DEBUG_CDB
+	bool "[DEPRECATED] Configuration database debug"
+	select DEPRECATED
+	depends on BT_MESH_CDB
+	help
+	  Use this option to enable configuration database debug logs.
+
+config BT_MESH_DEBUG_CFG
+	bool "[DEPRECATED] Configuration debug"
+	select DEPRECATED
+	help
+	  Use this option to enable node configuration debug logs for the
+	  Bluetooth mesh functionality.
+
+endmenu # [DEPRECATED] Mesh
+
+menu "[DEPRECATED] Services"
+
+# IAS
+
+config BT_DEBUG_IAS_CLIENT
+	bool "[DEPRECATED] Immediate Alert Service Client debug"
+	select DEPRECATED
+	depends on BT_IAS_CLIENT
+	help
+	  This option enables enables Immediate Alert Service Client debug logs.
+
+# OTS
+
+# TODO: Unify logging for OTS server and client - the client uses
+# "BT" debugging, the server does not.
+
+config BT_DEBUG_OTS_CLIENT
+	bool "[DEPRECATED] Object Transfer Service Client debug"
+	select DEPRECATED
+	help
+	  Use this option to enable Object Transfer Client debug logs
+
+endmenu # [DEPRECATED] Services
+
+endif # BT_DEPRECATED_LOG
+
+# (subsys/bluetooth/Kconfig)
+
+module = BT
+module-str = "Bluetooth"
+source "subsys/logging/Kconfig.template.log_config"
+
+# COMMON (subsys/bluetooth/common/Kconfig)
+
+module = BT_HCI_DRIVER
+legacy-debug-sym = BT_DEBUG_HCI_DRIVER
+module-str = "Bluetooth HCI driver"
+source "subsys/bluetooth/common/Kconfig.template.log_config_bt"
+
+module = BT_RPA
+legacy-debug-sym = BT_DEBUG_RPA
+module-str = "Bluetooth Resolvable Private Address (RPA)"
+source "subsys/bluetooth/common/Kconfig.template.log_config_bt"
+
+menu "Audio"
+
+# AICS
+
+module = BT_AICS
+legacy-debug-sym = BT_DEBUG_AICS
+module-str = "Audio Input Control Service"
+source "subsys/bluetooth/common/Kconfig.template.log_config_bt"
+
+module = BT_AICS_CLIENT
+legacy-debug-sym = BT_DEBUG_AICS_CLIENT
+module-str = "Audio Input Control Service client"
+source "subsys/bluetooth/common/Kconfig.template.log_config_bt"
+
+# BAP
+
+module = BT_BAP_STREAM
+legacy-debug-sym = BT_BAP_DEBUG_STREAM
+module-str = "Bluetooth Audio Stream"
+source "subsys/bluetooth/common/Kconfig.template.log_config_bt"
```

Comment:
How do we handle dependencies with these btw? 

Should it be possible to set `CONFIG_BT_BAP_STREAM_LOG_LEVEL_DBG=y` when `CONFIG_BT_BAP_STREAM=n`?

Reply:
IMO it's okay to have all the logging symbols available, even if the module they are used for isn't enabled.

It avoid having to use the `#define LOG_LEVEL ...` before the include of the logging system for files that reuse logging symbols.

Reply:
> It avoid having to use the #define LOG_LEVEL ... before the include of the logging system for files that reuse logging symbols.

I don't follow - How is that related to dependencies here?

Reply:
If you add a `depends on` on a symbol, this symbol will only be visible if the dependency is satisfied. For example in the legacy symbol, `BT_BAT_DEBUG_STREAM` depends on `BT_BAP_STREAM` which mean that if `BT_BAP_STREAM` isn't enabled, `BT_BAP_DEBUG_STREAM` will not be visible. That's the same if you guard the symbol definition with a `if`.

This is a problem if a file use that symbol to register the logging but does not always need the dependency to be used.

This lead to those kind of things:

https://github.com/zephyrproject-rtos/zephyr/blob/8bd0e13fc7d66a95194a9f5caf87daf8b469a949/subsys/bluetooth/audio/aics.c#L23-L25

Reply:
I'm sorry, I still do not follow. 

> If you add a depends on on a symbol, this symbol will only be visible if the dependency is satisfied. For example in the legacy symbol, BT_BAT_DEBUG_STREAM depends on BT_BAP_STREAM which mean that if BT_BAP_STREAM isn't enabled, BT_BAP_DEBUG_STREAM will not be visible. That's the same if you guard the symbol definition with a if.

This sounds related to 3rd party tools like `menuconfig` and doesn't apply to source files. 

> This is a problem if a file use that symbol to register the logging but does not always need the dependency to be used.
> 
> This lead to those kind of things:

In the example you give here, `CONFIG_BT_AICS` will always be enabled if the file is used, so the dependency here is always met. This should, IMO, always be the case. For example it should not, IMO, be possible to use `CONFIG_BT_BAP_STREAM_LOG_LEVEL_DBG` instead of `aics.c` if `CONFIG_BT_BAP_STREAM=n`.